File: src-Kotlin/BubbleSort.kt
import java.util.Scanner

class BubbleSort {

    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        System.out.print("Enter the number of elements : ")
        val n = sc.nextInt()
        val arr = IntArray(n)
        System.out.println("Enter $n elements :")
        for (i in 0 until n)
            arr[i] = sc.nextInt()

        bubbleSort(arr)

        System.out.println("\nThe sorted array : ;")
        for (i in 0 until n)
            System.out.print(arr[i].toString() + " ")
        System.out.println()
    }

    internal fun bubbleSort(arr: IntArray) {
        val len = arr.size
        var tmp: Int
        var flag: Boolean
        for (i in 0 until len) {
            flag = false
            for (j in 0 until len - i - 1) {
                if (arr[j] > arr[j + 1]) {
                    tmp = arr[j]
                    arr[j] = arr[j + 1]
                    arr[j + 1] = tmp
                    flag = true
                }
            }
            if (!flag)
                break
        }
    }
}

File: src-go/BubbleSort.go
package main

import (
	"fmt"
)

func bubbleSort(arr []int) {

	swapped := true;
	for swapped {
		swapped = false
		for i := 0; i < len(arr) - 1; i++ {
			if arr[i + 1] < arr[i] {
				Swap(arr, i, i + 1)
				swapped = true
			}
		}
	}	
}

func Swap(arr []int, i, j int) {
	tmp := arr[j]
	arr[j] = arr[i]
	arr[i] = tmp
}

func main() {

	arr := []int{1, 6, 2, 4, 9, 0, 5, 3, 7, 8}
	fmt.Println("Unsorted array: ", arr)
	bubbleSort(arr)
	fmt.Println("Sorted array: ", arr)
}